hw/ip/prim/dv/prim_ram_scr/cpp/scramble_model.cc

```diff
@@ -0,0 +1,373 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "scramble_model.h"
+#include <cassert>
+#include <iostream>
+#include <stdint.h>
+#include <vector>
+#include "prince_ref.h"
+
+uint8_t PRESENT_SBOX4[] = {0xc, 0x5, 0x6, 0xb, 0x9, 0x0, 0xa, 0xd,
+                           0x3, 0xe, 0xf, 0x8, 0x4, 0x7, 0x1, 0x2};
+
+uint8_t PRESENT_SBOX4_INV[] = {0x5, 0xe, 0xf, 0x8, 0xc, 0x1, 0x2, 0xd,
+                               0xb, 0x4, 0x6, 0x3, 0x0, 0x7, 0x9, 0xa};
+
+static const uint32_t kNumAddrSubstPermRounds = 2;
+static const uint32_t kNumDataSubstPermRounds = 2;
+static const uint32_t kNumPrinceHalfRounds = 2;
+
+static std::vector<uint8_t> byte_swap_vector(
+    const std::vector<uint8_t> &vec_in) {
```

Comment:
I think this is a byte reversal, so maybe `byte_reverse` would be a better name? To me, `byte_swap` sounds like it swaps 2 bytes, rather than all of them.